### PR TITLE
[cleanup] Remove stickiness filter

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- the split filter types is proving to be more of a headache than helpfui
- first step in removing separated filters

*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*
existing test suite should cover any changes

### CODE REVIEW

File: posthog/models/filters/mixins/common.py [hunk 1]
```diff
@@ -408,3 +408,13 @@ def target_entity_type(self) -> Optional[str]:
     @include_dict
     def entity_type_to_dict(self):
         return {"entity_type": self.target_entity_type} if self.target_entity_type else {}
+
+
+class SelectedIntervalMixin(BaseParamMixin):
```
Comment (macobo) [inline posthog/models/filters/mixins/common.py:413]: Should we perhaps clearly indicate with naming or comments that this is stickiness-specific?

File: posthog/models/filters/mixins/common.py [hunk 2]
```diff
@@ -408,3 +408,13 @@ def target_entity_type(self) -> Optional[str]:
...
+    @cached_property
+    def selected_interval(self) -> int:
```
Comment (macobo) [inline posthog/models/filters/mixins/common.py:415]: No tests for this anymore?

**Review by macobo — APPROVED** [review]
A few comments on tests/naming but otherwise great stuff!

**Comment by hazzadous** [comment]
> the split filter types is proving to be more of a headache than helpfui

Don't have much context on this change, what are the headaches? Looks like there's some stuff in e.g. the column optimizer that we wouldn't want there, but I'd suggest we can remove this in other ways that doesn't drop insight specific data structures, which I find personally useful. e.g. inheriting from something that includes all the types that column_optimizer expects, or passing in specific flags to column optimizer rather than the filter itself.

I would generally avoid trying to create a god data structure and introducing context specific things like `selected_interval` where it doesn't make sense. It makes it very difficult to see what is important for a specific insight, and makes typing also a little awkward e.g. we need to make everything optional when they may be required for certain requests.

@EDsCODE @macobo would love to understand more, happy to hangout.

**Comment by EDsCODE** [comment]
The headache was primarily from playing wack a mole with the typing weirdness. On one hand yes this is creating more of a god data structure but it will reduce the overriding that needs to happen in the query classes to ensure the class is recognizing the right filter

**Comment by EDsCODE** [comment]
@hazzadous had a brief call about this. Decided against making all these changes. Going to stick with the current split because at some point we'll want to have stronger definitions of each insight request and having the parameters already split as they are now will be easier to understand

**Pull request closed without merging**